examol/start/kmeans.py

```diff
@@ -1,16 +1,42 @@
 """Algorithms which select molecules based on k-means clustering"""
+import numpy as np
+from sklearn.cluster import KMeans
 from .base import Starter
 from examol.score.rdkit.descriptors import compute_morgan_fingerprints
+from scipy.spatial.distance import cdist
 
 
 class KMeansStarter(Starter):
     """Select structurally distinct molecules by picking molecules at the centers of clusters"""
 
     def _select(self, to_select: list[str], count: int) -> list[str]:
-        # Compute the fingerprints for each molecule
-        finger = [
-            compute_morgan_fingerprints(x) for x in to_select
-        ]
-
-        # Pick the molecules at the center of each cluster
-        raise NotImplementedError()
+        """
+        Input:
+            to_select: list of SMILES strings to select from
+            count: number of molecules to select
+        Output:
+            list of SMILES strings of selected molecules
+        """
+        if count > len(to_select):
```

Comment:
It's great that you added this! Could you move it to the `.select` method of the base class? It'll be good to have it for all starters.